presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/UnaliasSymbolReferences.java

```diff
@@ -369,7 +369,7 @@ public PlanNode visitLimit(LimitNode node, RewriteContext<Void> context)
         @Override
         public PlanNode visitDistinctLimit(DistinctLimitNode node, RewriteContext<Void> context)
         {
-            return new DistinctLimitNode(node.getId(), context.rewrite(node.getSource()), node.getLimit(), node.isPartial(), canonicalize(node.getHashSymbol()));
+            return new DistinctLimitNode(node.getId(), context.rewrite(node.getSource()), node.getLimit(), node.isPartial(), canonicalizeAndDistinct(node.getOutputSymbols()), canonicalize(node.getHashSymbol()));
```

Comment:
This should be `canonicalizeAndDistinct(node.getDistinctSymbols())`, since output symbols can include the hash.